src/test/java/com/unstoppabledomains/resolution/LibraryTest.java

```diff
@@ -0,0 +1,135 @@
+            e.printStackTrace();
+        }
+    }
+
+    @Test
+    public void namehash2() {
+        try {
+            final String hash = resolution.namehash("brad.crypto");
+            assertEquals(hash, "0x756e4e998dbffd803c21d23b06cd855cdc7a4b57706c95964a37e24b47c10fc9");
+        } catch (final NamingServiceException e) {
+            e.printStackTrace();
+        }
+    }
+
+    @Test
+    public void wrongDomainNamehash() {
+        try {
+            resolution.namehash("unsupported");
+        } catch (final NamingServiceException e) {
+            assertEquals(e.getCode(), NSExceptionCode.UnsupportedDomain);
+        }
+    }
+
+    @Test
+    public void addr() {
+        try {
+            String address = resolution.addr("brad.crypto", "eth");
+            assertEquals(address, "0x45b31e01AA6f42F0549aD482BE81635ED3149abb");
+        } catch(Exception e) {
+            e.printStackTrace();
+        }
+    }
+
+    @Test
+    public void wrongDomainAddr() {
+        try {
+            resolution.addr("unregistered.crypto", "eth");
+        } catch(NamingServiceException e) {
+            assertEquals(e.getCode(), NSExceptionCode.UnregisteredDomain);
+        }
+    }
+
+    @Test
+    public void UnknownCurrency() {
+        try {
+            resolution.addr("brad.crypto", "unknown");
+        } catch(NamingServiceException e) {
+            assertEquals(e.getCode(), NSExceptionCode.UnknownCurrency);
+        }
+    }
+
+    @Test
+    public void notConfiguredCurrency() {
+        try {
+            resolution.addr("brad.crypto", "dodge");
+        } catch(NamingServiceException e) {
+            assertEquals(e.getCode(), NSExceptionCode.UnknownCurrency);
+        }
+    }
+
+    @Test
+    public void ipfsHash() {
+        try {
+            String ipfsHash = resolution.ipfsHash("brad.crypto");
+            assertEquals(ipfsHash, "QmVJ26hBrwwNAPVmLavEFXDUunNDXeFSeMPmHuPxKe6dJv");
+        } catch(Exception e) {
+            e.printStackTrace();
+        }
+    }
+
+    @Test
+    public void ownerTest() {
+        try {
+            String owner = resolution.owner("brad.crypto");
+            assertEquals(owner, "0x020e7c546b1567ffc7f6202ca5f748533523dadc");
+        } catch(Exception e) {
+            e.printStackTrace();
+        }
+    }
+
+    @Test
+    public void ownerFailTest() {
+        try {
+            resolution.owner("unregistered.crypto");
+        }catch(NamingServiceException e) {
+            assertEquals(e.getCode(), NSExceptionCode.UnregisteredDomain);
+        }
+    }
+
+    @Test
+    public void noIpfsHash() {
+        try {
+            resolution.ipfsHash("unregistered.crypto");
+        } catch(NamingServiceException e) {
+            assertEquals(e.getCode(), NSExceptionCode.RecordNotFound);
+        }
+    }
+
+    @Test
+    public void noEmailRecord() {
+        try {
+            resolution.email("brad.crypto");
+        } catch(NamingServiceException e) {
```

Comment:
Suppose that `email` will not throw any exception. Will this test fail? In case of `jest` it will not so we use a special function to assert exception is thrown.